test/EKP/runtests-linear-model.jl

```diff
@@ -0,0 +1,137 @@
+    #                           "0.01 times its previous value.",
+    #                           "\nConsider reducing the EK time step."))
+    EKP.update_ensemble!(ekiobj, g_ens)
+
+    # restart EKObj and test step
+    ekiobj = EKP.EKObj(initial_ensemble, param_names,
+                       y_obs, Γy, Inversion())
+
+    # EKI iterations
+    for i in 1:N_iter
+        params_i = ekiobj.u[end]
+        g_ens = G(params_i')'
+        EKP.update_ensemble!(ekiobj, g_ens)
+    end
+
+    # EKI results: Test if ensemble has collapsed toward the true parameter values
+    eki_final_result = vec(mean(ekiobj.u[end], dims=1))
+    # @test norm(u_star - eki_final_result) < 0.5
+
+    # Plot evolution of the EKI particles
+    eki_final_result = vec(mean(ekiobj.u[end], dims=1))
+    gr()
+    p = plot(ekiobj.u[1][:,1], ekiobj.u[1][:,2], seriestype=:scatter)
+    plot!(ekiobj.u[end][:, 1],  ekiobj.u[end][:,2], seriestype=:scatter)
+    plot!([u_star[1]], xaxis="u1", yaxis="u2", seriestype="vline",
+          linestyle=:dash, linecolor=:red)
+    plot!([u_star[2]], seriestype="hline", linestyle=:dash, linecolor=:red)
+    savefig(p, "EKI_test.png")
+
+    ###
+    ###  Calibrate (2): Ensemble Kalman Sampleer
+    ###
+    # TODO: Uncomment L. 98-106 once EKS is implemented
+    eksobj = EKP.EKObj(initial_ensemble, param_names,
+                      y_obs, Γy,
+                      Sampler(prior_mean, prior_cov))
+
+    # EKS iterations
+    for i in 1:N_iter
+        params_i = eksobj.u[end]
+        g_ens = G(params_i')'
+        EKP.update_ensemble!(eksobj, g_ens)
+    end
+
+    # Plot evolution of the EKS particles
+    eks_final_result = vec(mean(eksobj.u[end], dims=1))
+    gr()
+    p = plot(eksobj.u[1][:,1], eksobj.u[1][:,2], seriestype=:scatter)
+    plot!(eksobj.u[end][:, 1],  eksobj.u[end][:,2], seriestype=:scatter)
+    plot!([u_star[1]], xaxis="u1", yaxis="u2", seriestype="vline",
+          linestyle=:dash, linecolor=:red)
+    plot!([u_star[2]], seriestype="hline", linestyle=:dash, linecolor=:red)
+    savefig(p, "EKS_test.png")
+
+    posterior_cov_inv = (A' * (Γy\A) + 1 * Matrix(I, n_par, n_par)/prior_cov)
+    ols_mean          = (A'*(Γy\A)) \ (A'*(Γy\y_obs))
+    posterior_mean    = posterior_cov_inv \ ( (A'*(Γy\A)) * ols_mean  + (prior_cov\prior_mean))
+
+    #### This tests correspond to:
+    # EKI provides a solution closer to the ordinary Least Squares estimate
+    @test norm(ols_mean - eki_final_result) < norm(ols_mean - eks_final_result)
+    # EKS provides a solution closer to the posterior mean
+    @test norm(posterior_mean - eks_final_result) < norm(posterior_mean - eki_final_result)
+
+    ##### I expect this test to make sense:
+    # In words: the ensemble covariance is still a bit ill-dispersed since the
+    # algorithm employed still does not include the correction term for finite-sized
+    # ensembles.
+    @test abs(sum(diag(posterior_cov_inv\cov(eksobj.u[end]))) - n_par) > 1e-5
```

Comment:
This is maybe more of a feature of the current algorithm as opposed to a test. e.g are we actually implying we fail this test if the covariance is not ill-dispersed?

Reply:
As discussed, we will improve the tests in a future PR, this one is mainly about implementing the EKS.